src/games/strategy/triplea/ai/AdvancedUtils.java

```diff
@@ -86,7 +86,7 @@ public static Route trimRouteBeforeFirstTerWithEnemyUnits(final Route route, fin
   public static List<Territory> getTerritoriesWithinXDistanceOfYMatchingZ(final GameData data, final Territory start,
       final int maxDistance, final Match<Territory> match) {
     return getTerritoriesWithinXDistanceOfYMatchingZAndHavingRouteMatchingA(data, start, maxDistance, match,
-        Match.ALWAYS_MATCH);
+        (Match<Territory>)Match.ALWAYS_MATCH);
```

Comment:
Since I parametized the `Match.ALWAYS_MATCH` constant, to `?` I need to cast it always, when it's needed...
I would recommend replacing this constant with a `Match<t> getAlwaysMatch(T o){return new Match<t>();}` or something similar, so that casting is no longer needed. Or we could simply remove this constant and have those methods calling `new AlwaysMatch<T>();`